flockoff/utils/git_utils.py

```
@@ -0,0 +1,140 @@
+    Returns:
+        The current commit hash as a string, or None if an error occurs (e.g., not a git repository).
+    """
+    try:
+        commit_hash = subprocess.check_output(
+            ['git', 'rev-parse', 'HEAD'],
+            cwd=repo_path,
+            stderr=subprocess.PIPE
+        ).strip().decode('utf-8')
+        return commit_hash
+    except subprocess.CalledProcessError as e:
+        warnings.warn(f"Error getting local commit hash: {e}. Ensure you are in a git repository.")
```

Ruff (0.8.2)

No explicit `stacklevel` keyword argument found

(B028)